src/main/java/ch/njol/skript/expressions/ExprIndicesOfX.java

```diff
@@ -0,0 +1,132 @@
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.SyntaxStringBuilder;
+import ch.njol.skript.lang.Variable;
+import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.skript.util.LiteralUtils;
+import ch.njol.util.Kleenean;
+import org.bukkit.event.Event;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+@Name("Indices of X in List")
+@Description("Returns all the indices of a list where the value at that index is the provided value.")
+@Examples({
+	"set {_list::*} to 1, 2, 3, 1, 2, 3",
+	"set {_indices::*} to the indices of the value 1 in {_list::*}",
+	"# {_indices::*} is now \"1\", \"4\"",
+	"",
+	"set {_indices::*} to the indices of the value 2 in {_list::*}",
+	"# {_indices::*} is now \"2\", \"5\"",
+	"",
+	"set {_otherlist::burb} to 100",
+	"set {_otherlist::burp} to 100",
+	"set {_indices::*} to the first index of the value 100 in {_otherlist::*}",
+	"# {_indices::*} is now \"burb\"",
+	"set {_indices::*} to the last index of the value 100 in {_otherlist::*}",
+	"# {_indices::*} is now \"burp\""
+})
+@Since("INSERT VERSION")
+public class ExprIndicesOfX extends SimpleExpression<String> {
+
+	static {
+		Skript.registerExpression(ExprIndicesOfX.class, String.class, ExpressionType.COMBINED,
+			"[the] [1:first|2:last] (indices|index[es]) of [[the] value] %object% in %objects%"
+		);
+	}
+
+	private IndexType type;
+	private Expression<?> value;
+	private Variable<?> list;
+
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		if (!(exprs[1] instanceof Variable<?> var)) {
+			Skript.error("'" + exprs[1].toString(null, false) + "' is not a list variable.");
+			return false;
+		}
+		if (!var.isList()) {
+			Skript.error("'" + exprs[1].toString(null, false) +
+					"' can only ever have one value at most, thus the 'indices of x in list' expression has no effect.");
+			return false;
+		}
+
+		list = var;
+		type = IndexType.values()[parseResult.mark];
+		value = LiteralUtils.defendExpression(exprs[0]);
+
+		return LiteralUtils.canInitSafely(value);
+	}
+
+	@Override
+	protected String @Nullable [] get(Event event) {
+		Object value = this.value.getSingle(event);
+		if (value == null)
+			return new String[0];
+
+		//noinspection unchecked
+		Map<String, Object> variable = (Map<String, Object>) list.getRaw(event);
+		if (variable == null)
+			return new String[0];
+
+		List<String> indices = new ArrayList<>();
+
+		for (Map.Entry<String, Object> entry : variable.entrySet()) {
+			Object entryValue = entry.getValue();
+			if (entryValue instanceof Map<?, ?> map)
+				entryValue = map.get(null);
```

Comment:
why are you checking this? this would include `{var::value}` in the list `{var::value::*}`

Reply:
huh, should it not be like this? e.g.
```
{list::1} -> 10
{list::1::1} -> 10
{list::2} -> 10
```
and do `indices of value 10 in {list::*}` would return `1` and `2`
if i were to not do that, it would only return `2`

Reply:
oh i see, I was thinking it was acting on the map one level up